lxd/container_lxc.go

```diff
@@ -3924,11 +3995,79 @@ func (c *containerLXC) Update(args db.ContainerArgs, userRequested bool) error {
 				if err != nil {
 					return err
 				}
-			} else if m["type"] == "nic" {
+			} else if m["type"] == "nic" || m["type"] == "infiniband" {
 				err = c.removeNetworkDevice(k, m)
 				if err != nil {
 					return err
 				}
+
+				// Wipe all per-{P,V}F unix devices associated
+				// with this infiniband device.
+				if m["type"] == "infiniband" {
+					// Check that we indeed have devices to remove
+					empty, _ := shared.PathIsEmpty(c.DevicesPath())
+					if empty {
+						return nil
```

Comment:
That sounds like a bad idea given you're in the middle of Update() here. Should that be "continue" perhaps?

Reply:
That's assuming we're in a loop, working with limited context here :)

Reply:
addressed